files-community/Files
# Code Quality: Improved UserControls codebase

## Summary

This namespace was one of the places where there were leftovers from UWP.

Re-organized codebase in UserControls to enhance maintainability and removed some XAML files(OpacityIcon and DataGridHeader), so it must be little but the app should be faster a bit.

## Task Checklist

- [x] Rename `IAddressToolBar` to `IAddressToolbarViewModel`
- [x] Rename `ISearchBox` to `ISearchBoxViewModel`
- [x] Rename `ISearchBox` to `ISearchBoxViewModel`
- [x] Rename `Views.LayoutModes` to `Views.ContentLayouts`
- [x] Rename `ViewModels.LayoutModes` to `ViewModels.ContentLayouts`
- [x] Move `IAddressToolbarViewModel` from `UserControls` to `ViewModels.UserControls`
- [x] Move `ISearchBoxViewModel` from `UserControls` to `ViewModels.UserControls`
- [x] Move `ViewModels.Previews` to `ViewModel.UserControls.Previews`
- [x] Move `ViewModels.Widgets` to `ViewModel.UserControls.Widgets`
- [x] Change base class of `DataGridHeader` from `UserControl` to `ButtonBase`
- [x] Ensure we folllow our codebase guideline

## Known Issues

Following issues would be fixed in the upcoming PR.

_N/A_

## PR Checklist

- [x] **Related**: #4180

## Steps To Validate Changes

There're some classes that were changed codebase largely. Please have look at:

- See that DataGridHeader is working well
- See that OpacityIcon is working well


## Screenshots

_N/A_

[^1]: If the request hasn't been agreed by the team, this work might be rejected. Make sure that you get approval from the team before opening any PR related the request.
[^2]: If you removed any en-US string resources, make sure that there are no references of those resources. When you add a new en-US string resources, do not make any changes on other languages' resources; [Crowdin](https://crowdin.com/project/files-app) will do, instead.
